src/debuggability.js

```diff
@@ -141,6 +141,63 @@ Promise.hasLongStackTraces = function () {
     return config.longStackTraces && longStackTracesIsSupported();
 };
 
+var supportedEvents = ["created", "chained", "fulfilled",
+    "rejected", "resolved", "cancelled"];
+
+Promise.on = function (eventName, hookFunction, simpleEventApi) {
+    if (supportedEvents.indexOf(eventName) < 0) {
+        throw new Error("You can only subscribe to these events:" +
+            supportedEvents);
+    }
+    var newHook = null;
+    if (simpleEventApi) {
+        // This interface allows users to subscribe to promises
+        // related events in more performable way then event API
+        // hookFunction is called after the event it has this
+        // (current promise reference) defined and single
+        // argument (child promise in "chained" event)
+        newHook = hookFunction;
+    } else {
+        newHook = function (other) {
+            hookFunction({
+                guid: this._promiseId ? this._promiseId : null,
+                childGuid: (other && other._promiseId) ?
+                    other._promiseId : null,
+                eventName: eventName,
+                detail: this.isRejected() ? this.reason :
+                    this.isFulfilled() ? this.value : null,
+                //event.label //is this relevant?
+                timeStamp: Date.now(),
+                stack: this._trace ? this._trace : null
+            });
+        };
+        hookFunction._eventHandler = newHook;
+    }
+    util.hookTo(Promise.prototype, "_hook_" + eventName, newHook);
+};
+
+Promise.off = function (eventName, hookFunction) {
+    if (supportedEvents.indexOf(eventName) < 0) {
+        throw new Error("You can only subscribe to these events:"
+            + supportedEvents);
+    }
+    util.unhookFrom(Promise.prototype, "_hook_" + eventName,
+        hookFunction._eventHandler ? hookFunction._eventHandler :
+            hookFunction);
+};
+
+// Event handlers can be placed on (window|process).__PROMISE_INSTRUMENTATION__
```

Comment:
I think firing global events is more complex in terms of maintainability and performance.
Regarding \__PROMISE_INSTRUMENTATION\__: as far as I understand @petkaantonov wanted the events to be fired same in same way as in RSVP, here https://github.com/tildeio/rsvp.js/blob/fcd59034548508a2126b9a3cd1c7132288a7b04f/lib/rsvp.js#L39 I have found sngle reference to window in context of events in RSVP code. From RSVP code I understand that library users are expected to put the event handlers on window.\__PROMISE_INSTRUMENTATION\__ properties, this is the way in RSVP I did it the same. Of course there is an option to fire DOM events (performance?) or to call window.onSomethingHappened() methods (name compatibility), as it is done in debuggability.js:fireGlobalEvent() but these methods look less convenient to me.